test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/TestServer.cs

```diff
@@ -47,63 +49,68 @@ public TestServer(RequestDelegate app, TestServiceContext context, ListenOptions
 
         public TestServer(RequestDelegate app, TestServiceContext context, ListenOptions listenOptions, IHttpContextFactory httpContextFactory)
         {
+            _app = app;
             _listenOptions = listenOptions;
-
             Context = context;
-            context.ServerOptions.ListenOptions.Add(_listenOptions);
-
-            // Switch this to test on socket transport
-            var transportFactory = CreateLibuvTransportFactory(context);
-            // var transportFactory = CreateSocketTransportFactory(context);
-
-            _server = new KestrelServer(transportFactory, context);
-            var httpApplication = new DummyApplication(app, httpContextFactory);
-
-            try
-            {
-                _server.StartAsync(httpApplication, CancellationToken.None).Wait();
-            }
-            catch
-            {
-                _server.StopAsync(new CancellationTokenSource(_shutdownTimeout).Token).Wait();
-                _server.Dispose();
-                throw;
-            }
-        }
 
-        private static ITransportFactory CreateLibuvTransportFactory(TestServiceContext context)
-        {
-            var transportOptions = new LibuvTransportOptions { ThreadCount = 1 };
-
-            var transportFactory = new LibuvTransportFactory(
-                Options.Create(transportOptions),
-                new LifetimeNotImplemented(),
-                new KestrelTestLoggerFactory(new TestApplicationErrorLogger()));
-
-            return transportFactory;
+            _host = new WebHostBuilder()
+                     .UseKestrel(o =>
+                     {
+                         o.ListenOptions.Add(_listenOptions);
+                     })
+                     .ConfigureServices(services =>
```

Comment:
Before, IStartup.ConfigureServices and yes you need both. You need to replace some HostingServices and you need to replace the `IApplicationLifetime` in IStartup.ConfigureServices because that's where it's added (we should have called TryAdd in hosting).